src/stream/src/executor/managed_state/dynamic_filter.rs

```diff
@@ -187,13 +188,22 @@ impl<S: StateStore> RangeCache<S> {
     /// owned.
     pub fn update_vnodes(&mut self, new_vnodes: Arc<Bitmap>) -> Arc<Bitmap> {
         let old_vnodes = self.state_table.update_vnode_bitmap(new_vnodes.clone());
-        for (vnode, (old, new)) in old_vnodes.iter().zip_eq(new_vnodes.iter()).enumerate() {
-            if old && !new {
-                let vnode = vnode.try_into().unwrap();
-                self.cache.remove(&vnode);
+
+        // if new vnodes is not subset of old vnodes, clear cache and range, else delete the stale
+        // vnodes
+        if cache_may_stale(old_vnodes.as_ref(), new_vnodes.as_ref()) {
```

Comment:
Is it possible to have other problems with this modification? cc @BugenZhao @jon-chuang

Reply:
I guess this works.